aiohttp/connector.py

```diff
@@ -464,11 +510,25 @@ def _create_connection(self, req):
 
         for hinfo in hosts:
             try:
-                return (yield from self._loop.create_connection(
-                    self._factory, hinfo['host'], hinfo['port'],
+                host = hinfo['host']
+                port = hinfo['port']
+                conn = yield from self._loop.create_connection(
+                    self._factory, host, port,
                     ssl=sslcontext, family=hinfo['family'],
                     proto=hinfo['proto'], flags=hinfo['flags'],
-                    server_hostname=hinfo['hostname'] if sslcontext else None))
+                    server_hostname=hinfo['hostname'] if sslcontext else None)
+                if req.ssl and self._expect_fingerprint:
```

Comment:
Sorry for bad guess, `req.ssl` may clash with `ProxyConnector`.

Please use `transport.get_extra_info('sslcontext')`  instead as flag that `transport.get_extra_info('socket')` has certificate info.